signer/utils.go

```diff
@@ -0,0 +1,109 @@
+package signer
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+)
+
+// DeriveSignatureAlgorithm picks up a recommended signing algorithm for given certificate.
+func DeriveSignatureAlgorithm(signingCert *x509.Certificate) (SignatureAlgorithm, error) {
+	keySpec, err := DeriveKeySpec(signingCert)
+	if err != nil {
+		return "", err
+	}
+
+	return keySpec.SignatureAlgorithm(), nil
+}
+
+// DeriveKeySpec picks up a recommended signing algorithm for given certificate.
+func DeriveKeySpec(signingCert *x509.Certificate) (KeySpec, error) {
```

Comment:
Can this be a private method?

Reply:
Its an library we want this helper method to be public for consumers to derive keyspec from cert